### PR TITLE
[show/config]: combine feature and container feature cli

### PR DESCRIPTION
**Pull request by lguohan**
Signed-off-by: Guohan Lu <lguohan@gmail.com>

**- What I did**
merge container feature cli into feature cli
add unit tests

```
02:24:22  ============================= test session starts ==============================
02:24:22  platform linux2 -- Python 2.7.16, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
02:24:22  rootdir: /sonic/sonic-utilities, inifile: pytest.ini
02:24:22  plugins: cov-2.6.0
02:24:24  collected 75 items
02:24:24  
02:24:24  sonic-utilities-tests/acl_loader_test.py ....                            [  5%]
02:24:24  sonic-utilities-tests/aclshow_test.py ...........                        [ 20%]
02:24:25  sonic-utilities-tests/config_mgmt_test.py ....                           [ 25%]
02:24:26  sonic-utilities-tests/drops_group_test.py .......                        [ 34%]
02:24:26  sonic-utilities-tests/feature_test.py .........                          [ 46%]
02:24:27  sonic-utilities-tests/filter_fdb_entries_test.py ..........              [ 60%]
02:24:27  sonic-utilities-tests/gearbox_test.py ..                                 [ 62%]
02:24:36  sonic-utilities-tests/intfstat_test.py ........                          [ 73%]
02:24:39  sonic-utilities-tests/intfutil_test.py ........                          [ 84%]
02:24:39  sonic-utilities-tests/port2alias_test.py ....                            [ 89%]
02:24:40  sonic-utilities-tests/psu_test.py ...                                    [ 93%]
02:24:41  sonic-utilities-tests/sfp_test.py ...                                    [ 97%]
02:24:41  sonic-utilities-tests/show_breakout_test.py ..                           [100%]
```

config command:
```
admin@vlab-01:~$ sudo config feature
Usage: config feature [OPTIONS] COMMAND [ARGS]...

  Modify configuration of features

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  autorestart  Configure autorestart status for a feature
  state        Configure status of feature
```

show command:
```
admin@vlab-01:~$ show feature
Usage: show feature [OPTIONS] COMMAND [ARGS]...

  Show feature status

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  autorestart  Show auto-restart status for a feature
  status       Show feature status
```

output:

```
admin@vlab-01:~$ show feature status
Feature     State     AutoRestart
----------  --------  -------------
bgp         enabled   enabled
database    enabled   disabled
dhcp_relay  enabled   enabled
lldp        enabled   enabled
nat         enabled   enabled
pmon        enabled   enabled
radv        enabled   enabled
restapi     disabled  enabled
sflow       disabled  enabled
snmp        enabled   enabled
swss        enabled   enabled
syncd       enabled   enabled
teamd       enabled   enabled
telemetry   enabled   enabled
admin@vlab-01:~$ show feature autorestart
Feature     AutoRestart
----------  -------------
bgp         enabled
database    disabled
dhcp_relay  enabled
lldp        enabled
nat         enabled
pmon        enabled
radv        enabled
restapi     enabled
sflow       enabled
snmp        enabled
swss        enabled
syncd       enabled
teamd       enabled
telemetry   enabled
```
**- How I did it**
There will be other changes needed and unit added.

This is just to get some feedback about the cli.

**- How to verify it**
validate in virtual switch.

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -3770,12 +3770,20 @@ def delete(ctx):
...
+
+#
+# 'feature' command ('config feature state ...')
+#
+@feature.command('state')
 @click.argument('name', metavar='<feature-name>', required=True)
 @click.argument('state', metavar='<feature-state>', required=True, type=click.Choice(["enabled", "disabled"]))
-def feature_status(name, state):
+def feature_state(name, state):
     """ Configure status of feature"""
```
Comment (jleveque) [inline config/main.py:3787]: s/status/state/ here to align with new subcommand name

File: config/main.py [hunk 2]
```diff
@@ -3770,12 +3770,20 @@ def delete(ctx):
...
-@config.command('feature')
+# 'feature' group ('config feature ...')
+#
+@config.group(cls=AbbreviationGroup, name='feature', invoke_without_command=False)
+def feature():
+    """Modify configuration of features"""
+    pass
+
+#
+# 'feature' command ('config feature state ...')
```
Comment (jleveque) [inline config/main.py:3781]: Change to `# 'state' subcommand ('config feature state ...')`

File: show/main.py [hunk 3]
```diff
@@ -3051,53 +3051,46 @@ def ztp(status, verbose):
...
-@container.group(name='feature', invoke_without_command=False)
-def feature():
-    """Show container feature"""
-    pass
+@feature.command('status', short_help="Show feature status")
+@click.argument('feature_name', required=False)
+def autorestart(feature_name):
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    header = ['Feature', 'Status', 'AutoRestart']
```
Comment (jleveque) [inline show/main.py:3069]: 'Status' should change to 'State' to align with terminology from `config` command

**Comment by lguohan** [comment]
@jleveque , unit tests added, please take a look.

File: sonic-utilities-tests/feature_test.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+bgp        disabled
+"""
+
+class TestFeature(object):
+    @classmethod
+    def setup_class(cls):
+        print("SETUP")
+
+    def test_show_feature_status(self):
+        runner = CliRunner()
```
Comment (jleveque) [inline sonic-utilities-tests/feature_test.py:92]: Suggest moving this instantialization to a setUp method and only performing it once:

```python
    def setUp(self):
        self.runner = CliRunner()
```

File: show/main.py [hunk 5]
```diff
@@ -3420,4 +3421,8 @@ def tunnel():
     click.echo(tabulate(table, header))
 
 if __name__ == '__main__':
+
```
Comment (jleveque) [inline show/main.py:3424]: Delete this empty line?

File: sonic-utilities-tests/feature_test.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
+import sys
+import os
+import mock
+from click.testing import CliRunner
+from swsssdk import ConfigDBConnector
```
Comment (jleveque) [inline sonic-utilities-tests/feature_test.py:5]: Suggest to reorganize imports per [PEP8 standards](https://www.python.org/dev/peps/pep-0008/#imports):

>Imports should be grouped in the following order:
>
>Standard library imports.
>Related third party imports.
>Local application/library specific imports.
>You should put a blank line between each group of imports.

```
import os
import sys

import mock
from click.testing import CliRunner
from swsssdk import ConfigDBConnector
```

File: sonic-utilities-tests/feature_test.py (hunk 4 above)
Comment (lguohan) [reply sonic-utilities-tests/feature_test.py:92]: setUp method is defined as class method. we cannot put it there.

Comment (lguohan) [reply sonic-utilities-tests/feature_test.py:92]: setUp is not the pytest way to run it, setup is old python unittest way, py.test way is to use fixture. actually, all setUp method should be clean up in previous test and we need to add fixture to do it. I suggest to do this in another pr. won't plan to fix this in this pr.

**Comment by lguohan** [comment]
@jleveque , rebased to current mastre

File: config/main.py [hunk 7]
```diff
@@ -890,6 +883,17 @@ def validate_mirror_session_config(config_db, session_name, dst_port, src_port,
 @click.group(cls=AbbreviationGroup, context_settings=CONTEXT_SETTINGS)
 def config():
     """SONiC command line - 'config' command"""
+    #
+    # Load asic_type for further use
+    #
+    try:
+        version_info = sonic_device_util.get_sonic_version_info()
+        asic_type = version_info['asic_type']
```
Comment (jleveque) [inline config/main.py:891]: Should you now declare asic_type as `global` first in this function?

Comment (lguohan) [reply config/main.py:893]: you are right, it should be global.

**Comment by lguohan** [comment]
test failure is not related to the pr.

**Pull request merged**